feat: workflow logs with console output support

shahargl opened the pull request:
## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

shahargl committed:
feat: workflow logs with console output support

shahargl committed:
fix: restore workflow log search functionality

- Created full WorkflowEventLoggerService with ES search capabilities
- Restored getExecutionLogs and getStepLogs with actual ES queries
- Fixed workflows management service to use proper logger service
- Maintained console logging functionality while adding back log search

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324678)
* Commit: 7c4b8a3d2c02e08edfac91b2378fe67220eac72a

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/324678#01985b01-829a-4b95-9ddf-294909c0412a)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324678","number":324678,"commit":"7c4b8a3d2c02e08edfac91b2378fe67220eac72a"}],"number":324678}
buildkite-pr-comment-->

shahargl committed:
fix: eliminate circular dependency in workflow logging

- Remove imports from workflows_management to workflows_execution_engine
- Implement direct ES log search in management service
- Keep simple console logging without complex service layer
- Clean plugin separation with no circular dependencies

shahargl committed:
fix: resolve undefined @timestamp error in workflow execution logs

- Fix data structure access: log['@timestamp'] instead of log.source['@timestamp']
- Add defensive filtering to remove undefined/null log entries
- Improve error handling in both service and API layers
- Handle ES responses that may have missing _source data

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

shahargl committed:
Merge branch 'main' into feature/workflow-logs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324709)
* Commit: 95a65453e45017b3e1b42e2308dcc73eb09c358d

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/324709#01985b38-2136-448e-9858-b750fcf8d43b)

### History
* :broken_heart: [Build #324693](https://buildkite.com/elastic/kibana-pull-request/builds/324693) failed f934c08236b1b6f129455a2ec2bc2083612cefa4
* :broken_heart: [Build #324678](https://buildkite.com/elastic/kibana-pull-request/builds/324678) failed 7c4b8a3d2c02e08edfac91b2378fe67220eac72a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324709","number":324709,"commit":"95a65453e45017b3e1b42e2308dcc73eb09c358d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324693","number":324693,"commit":"f934c08236b1b6f129455a2ec2bc2083612cefa4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324678","number":324678,"commit":"7c4b8a3d2c02e08edfac91b2378fe67220eac72a"}],"number":324709}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324720)
* Commit: 3a65903d008b7b6ba8ea97260a4b2b0212b64bf2

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/324720#01985b4f-7879-463f-afe1-b46ff1929e81)

### History
* :broken_heart: [Build #324709](https://buildkite.com/elastic/kibana-pull-request/builds/324709) failed 95a65453e45017b3e1b42e2308dcc73eb09c358d
* :broken_heart: [Build #324693](https://buildkite.com/elastic/kibana-pull-request/builds/324693) failed f934c08236b1b6f129455a2ec2bc2083612cefa4
* :broken_heart: [Build #324678](https://buildkite.com/elastic/kibana-pull-request/builds/324678) failed 7c4b8a3d2c02e08edfac91b2378fe67220eac72a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324720","number":324720,"commit":"3a65903d008b7b6ba8ea97260a4b2b0212b64bf2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324709","number":324709,"commit":"95a65453e45017b3e1b42e2308dcc73eb09c358d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324693","number":324693,"commit":"f934c08236b1b6f129455a2ec2bc2083612cefa4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324678","number":324678,"commit":"7c4b8a3d2c02e08edfac91b2378fe67220eac72a"}],"number":324720}
buildkite-pr-comment-->

shahargl committed:
Merge branch 'main' into feature/workflow-logs

shahargl committed:
feat: workflow logs

shahargl committed:
feat: Merge branch 'feature/workflow-logs' of https://github.com/shahargl/kibana into feature/workflow-logs

shahargl committed:
feat: workflow logger

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324778)
* Commit: 297ded4963314d17dc9b4864f2608ada52e88518

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/324778#01985b96-2256-48cc-b4e0-e5ab7d7c78a7)

### History
* :broken_heart: [Build #324755](https://buildkite.com/elastic/kibana-pull-request/builds/324755) failed 44c2f5e2bff50c8ccf516c6d4c2a998fe387c6c5
* :broken_heart: [Build #324720](https://buildkite.com/elastic/kibana-pull-request/builds/324720) failed 3a65903d008b7b6ba8ea97260a4b2b0212b64bf2
* :broken_heart: [Build #324709](https://buildkite.com/elastic/kibana-pull-request/builds/324709) failed 95a65453e45017b3e1b42e2308dcc73eb09c358d
* :broken_heart: [Build #324693](https://buildkite.com/elastic/kibana-pull-request/builds/324693) failed f934c08236b1b6f129455a2ec2bc2083612cefa4
* :broken_heart: [Build #324678](https://buildkite.com/elastic/kibana-pull-request/builds/324678) failed 7c4b8a3d2c02e08edfac91b2378fe67220eac72a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324778","number":324778,"commit":"297ded4963314d17dc9b4864f2608ada52e88518"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324755","number":324755,"commit":"44c2f5e2bff50c8ccf516c6d4c2a998fe387c6c5"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324720","number":324720,"commit":"3a65903d008b7b6ba8ea97260a4b2b0212b64bf2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324709","number":324709,"commit":"95a65453e45017b3e1b42e2308dcc73eb09c358d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324693","number":324693,"commit":"f934c08236b1b6f129455a2ec2bc2083612cefa4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324678","number":324678,"commit":"7c4b8a3d2c02e08edfac91b2378fe67220eac72a"}],"number":324778}
buildkite-pr-comment-->

shahargl committed:
feat: workflows

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324809)
* Commit: 148b4aa26f08f76d9dd83fd57042a9b3142fc139

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/324809#01985bbb-d496-40a6-8496-368a525fcfd4)

### History
* :broken_heart: [Build #324778](https://buildkite.com/elastic/kibana-pull-request/builds/324778) failed 297ded4963314d17dc9b4864f2608ada52e88518
* :broken_heart: [Build #324755](https://buildkite.com/elastic/kibana-pull-request/builds/324755) failed 44c2f5e2bff50c8ccf516c6d4c2a998fe387c6c5
* :broken_heart: [Build #324720](https://buildkite.com/elastic/kibana-pull-request/builds/324720) failed 3a65903d008b7b6ba8ea97260a4b2b0212b64bf2
* :broken_heart: [Build #324709](https://buildkite.com/elastic/kibana-pull-request/builds/324709) failed 95a65453e45017b3e1b42e2308dcc73eb09c358d
* :broken_heart: [Build #324693](https://buildkite.com/elastic/kibana-pull-request/builds/324693) failed f934c08236b1b6f129455a2ec2bc2083612cefa4
* :broken_heart: [Build #324678](https://buildkite.com/elastic/kibana-pull-request/builds/324678) failed 7c4b8a3d2c02e08edfac91b2378fe67220eac72a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324809","number":324809,"commit":"148b4aa26f08f76d9dd83fd57042a9b3142fc139"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324778","number":324778,"commit":"297ded4963314d17dc9b4864f2608ada52e88518"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324755","number":324755,"commit":"44c2f5e2bff50c8ccf516c6d4c2a998fe387c6c5"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324720","number":324720,"commit":"3a65903d008b7b6ba8ea97260a4b2b0212b64bf2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324709","number":324709,"commit":"95a65453e45017b3e1b42e2308dcc73eb09c358d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324693","number":324693,"commit":"f934c08236b1b6f129455a2ec2bc2083612cefa4"}],"number":324809}
buildkite-pr-comment-->

shahargl committed:
feat: workflows

shahargl committed:
feat: workflows

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/325017)
* Commit: 72fa292f13814acddb98af35c4e7c2d532506e3b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #324989](https://buildkite.com/elastic/kibana-pull-request/builds/324989) failed a4757cfcee556db58bb33adcf735185a1142aeca
* :broken_heart: [Build #324809](https://buildkite.com/elastic/kibana-pull-request/builds/324809) failed 148b4aa26f08f76d9dd83fd57042a9b3142fc139
* :broken_heart: [Build #324778](https://buildkite.com/elastic/kibana-pull-request/builds/324778) failed 297ded4963314d17dc9b4864f2608ada52e88518
* :broken_heart: [Build #324755](https://buildkite.com/elastic/kibana-pull-request/builds/324755) failed 44c2f5e2bff50c8ccf516c6d4c2a998fe387c6c5
* :broken_heart: [Build #324720](https://buildkite.com/elastic/kibana-pull-request/builds/324720) failed 3a65903d008b7b6ba8ea97260a4b2b0212b64bf2

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/325017","number":325017,"commit":"72fa292f13814acddb98af35c4e7c2d532506e3b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324989","number":324989,"commit":"a4757cfcee556db58bb33adcf735185a1142aeca"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324809","number":324809,"commit":"148b4aa26f08f76d9dd83fd57042a9b3142fc139"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324778","number":324778,"commit":"297ded4963314d17dc9b4864f2608ada52e88518"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324755","number":324755,"commit":"44c2f5e2bff50c8ccf516c6d4c2a998fe387c6c5"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324720","number":324720,"commit":"3a65903d008b7b6ba8ea97260a4b2b0212b64bf2"}],"number":325017}
buildkite-pr-comment-->
